### DISCUSSION TITLE
About `org` `app` setting in `serverless.yml` file(serverless framework 4.1.12)

### DISCUSSION THREAD

**Discussion by shiki-sou** [serverless/serverless#12684] [Q&A]
Our service can be deployed when we set both 'org' and 'app'.

When we deploy our service with `serverless.yml` file setting below.
``` yml
service: ourService
org: ourOrg
# app: ourApp
provider:
  name: aws
```
``` sh
export SERVERLESS_ACCESS_KEY=AKxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
serverless deploy --config serverless.yml --stage xxx
```
We get error message below
``` text
(node:17488) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)

✖ AWS credentials missing or invalid. Run "serverless" to set up AWS credentials, or learn more in our docs: http://slss.io/aws-creds-setup. Original error from AWS: Could not load credentials from any providers
```

When we deploy our service with `serverless.yml` file setting below.
``` yml
service: ourService
# org: ourOrg
# app: ourApp
provider:
  name: aws
```
``` sh
export SERVERLESS_ACCESS_KEY=AKxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
serverless deploy --config serverless.yml --stage xxx
```
We get error message below
``` text
(node:17730) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)

✖ AWS credentials missing or invalid. Run "serverless" to set up AWS credentials, or learn more in our docs: http://slss.io/aws-creds-setup. Original error from AWS: Could not load credentials from any providers
```

From the link below, in comment's description, we found that 'org' and 'app' is optional.
https://www.serverless.com/framework/docs/providers/aws/guide/serverless.yml#root-properties

So is there any way we could deploy our service without 'org' 'app' setting?
The reason we don't want to set 'org' 'app' is that we don't want to use dashboard feature since it will spend extra credit.

**Comment by garethmcc** [comment]
Hi there. Neither the org nor app are required. They are only required if you wish to make use of Serverless Dashboard services such as Providers, Parameters and CI/CD. 

If you do not use org and app settings, however, you cannot use the connection to AWS provided by Providers in your Serverless Dashboard account and you will need to have AWS credentials local to your deployment environment. This can be configured in many ways standard to AWS such as creating access key and secret on a user in AWS and then assigning them as environment variables (https://docs.aws.amazon.com/cli/v1/userguide/cli-configure-envvars.html#envvars-set)

**Reply by shiki-sou** [reply]
We tried below setting as aws's guide book wrote.
``` sh
export SERVERLESS_ACCESS_KEY=AKxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
export AWS_ACCESS_KEY_ID=XXXXXXXXXXXXXXXXXXXXXXXXX
export AWS_SECRET_ACCESS_KEY=XXXXXXXXXXXXXXXXXXXXXXXXX
export AWS_DEFAULT_REGION=XXXXXXXX

serverless deploy --config serverless.yml --stage xxx
```
It still give us same error message
``` text
(node:22889) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)

✖ AWS credentials missing or invalid. Run "serverless" to set up AWS credentials, or learn more in our docs: http://slss.io/aws-creds-setup. Original error from AWS: Could not load credentials from any providers
```

**Reply by Mmarzex** [reply]
@shiki-sou try removing the `SERVERLESS_ACCESS_KEY` env and leaving the AWS credential variables.
